xarray/backends/plugins.py

```diff
@@ -135,6 +135,8 @@ def guess_engine(
         try:
             if backend.guess_can_open(store_spec):
                 return engine
+        except PermissionError:
+            raise
```

Comment:
```suggestion
        except PermissionError as e:
            raise e
```


I think this will show the user the actual error message but not sure. Can you test it out?

Reply:
The full error is already there. If you don't specify what to raise in an caught exception, it raises the original again. See the description of this Pull Request which shows that.